stream.go

```diff
@@ -0,0 +1,145 @@
+package tarantool
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+const (
+	// By default, the isolation level of Tarantool is serializable.
+	defaultIsolationLevel = iota
+	// The ReadCommittedLevel isolation level makes visible all transactions
+	// that started commit (stream.Do(NewCommitRequest()) was called).
+	ReadCommittedLevel
+	// The ReadConfirmedLevel isolation level makes visible all transactions
+	// that finished the commit (stream.Do(NewCommitRequest()) was returned).
+	ReadConfirmedLevel
+	// If the BestEffortLevel (serializable) isolation level becomes unreachable,
+	// the transaction is marked as «conflicted» and can no longer be committed.
+	BestEffortLevel
+)
+
+type Stream struct {
+	Id   uint64
+	Conn *Connection
+}
+
+func fillBegin(enc *msgpack.Encoder, txnIsolation uint64, timeout time.Duration) error {
+	hasTimeout := timeout > 0
+	hasIsolationLevel := txnIsolation != defaultIsolationLevel
+	mapLen := 0
+	if hasTimeout {
+		mapLen += 1
+	}
+	if hasIsolationLevel {
+		mapLen += 1
+	}
+
+	err := enc.EncodeMapLen(mapLen)
+
+	if hasTimeout {
+		enc.EncodeUint64(KeyTimeout)
+		err = enc.Encode(timeout.Seconds())
+	}
+
+	if hasIsolationLevel {
+		enc.EncodeUint64(KeyTxnIsolation)
+		err = enc.Encode(txnIsolation)
+	}
+
+	return err
+}
+
+func fillCommit(enc *msgpack.Encoder) error {
+	return enc.EncodeMapLen(0)
+}
+
+func fillRollback(enc *msgpack.Encoder) error {
+	return enc.EncodeMapLen(0)
+}
+
+// BeginRequest helps you to create an execute request object for execution
+// by a Connection.
+type BeginRequest struct {
```

Comment:
An user can execute BeginRequest, RollbackRequest and CommitRequest on a regular `Connection` and `ConnectionPool`. It may confuses.

Reply:
If user execute BeginRequest (for example), he gets an error `Failed to Begin: Unable to process BEGIN request out of stream (0xe6)`. Do you mean this situation needs to be highlighted in the documentation?

Reply:
Thank you, it's ok to validate it in the server-side.

It would be great to highlight the behavior in the comments for these requrests.

Reply:
Now there are comments like that:

https://github.com/tarantool/go-tarantool/blob/3a86a3f9c38290154195b716d207ae9493335797/stream.go#L67